l10n_eu_adr_report/report/dangerous_delivery_report.py

```diff
@@ -0,0 +1,66 @@
+# Copyright 2019 Iryna Vyshnevska (Camptocamp)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+from odoo import api, models
+
+
+class DangerousDeliveryADR(models.AbstractModel):
+    _name = (
+        'report.l10n_eu_adr_report.report_delivery_dangerous'
+    )
+
+    @api.model
+    def _get_report_values(self, docids, data=None):
+        docs = self.env['stock.picking']
+        data = data if data is not None else {}
+        picks = docs.browse(docids)
+        lines = self._prepare_dangerous_lines(picks)
+        docargs = {
+            'doc_ids': docs.ids,
+            'doc_model': 'stock.picking',
+            'docs': picks,
+            'data': data.get('form', False),
+            # Amount of 7 lines is to satisfy requirements for first page
+            # as this amount is fiting A4 page
+            'first_page_lines': lines[:7],
+            'next_page_lines': lines[7:],
+        }
+        return docargs
+
+    def _prepare_dangerous_lines(self, pickings):
+        vals = []
+        for pick in pickings:
+            for move_line in pick.move_line_ids:
```

Comment:
Maybe the two for loops can be merged into one with the mapped function ?

Reply:
we do not support printing on a bunch of picking for now anyway, can stay with one iteration